### DISCUSSION TITLE
Target-based Type inference for `new[] {...}` Array.

### DISCUSSION THREAD

**Discussion by las-nsc** [dotnet/csharplang#6175] [Language Ideas] [answered]
When both of these are possible:
`var thing = new Thing();`
`Thing thing = new();`
it seems like both of these should be possible too:
`var things = new[] { new Thing() };`
`Thing[] things = new[] { new() };`

The last one isn't currently possible, because `new[]` tries to find a common type based on all the elements contained. It should be easier and arguably more useful to just assume the target type is the desired one rather than trying to work it out from the contents. If the target type is `var` or unknown, then it should default back to the current behaviour.

**Comment by Neme12** [comment, marked as answer]
Possibly a duplicate of #2701
